testcases/OpTestKexec.py

```diff
@@ -0,0 +1,129 @@
+# permissions and limitations under the License.
+#
+# IBM_PROLOG_END_TAG
+#
+
+'''
+OpTestKexec
+---------
+
+This test is to validate kexec commands
+'''
+
+import os
+import pexpect
+import unittest
+import time
+import OpTestLogger
+
+import OpTestConfiguration
+from common.OpTestSystem import OpSystemState
+from common.Exceptions import CommandFailed
+from common.OpTestError import OpTestError
+
+log = OpTestLogger.optest_logger_glob.get_logger(__name__)
+
+class OpTestKexec(unittest.TestCase):
+
+    def setUp(self):
+        conf = OpTestConfiguration.conf
+        self.cv_SYSTEM = conf.system()
+        self.c = self.cv_SYSTEM.console
+        self.kernel_image = conf.args.kernel_image
+        self.initrd_image = conf.args.initrd_image
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        res = self.c.run_command("cat /etc/os-release")
+        if "Ubuntu" in (res[0] or res[1]):
+            self.distro = "ubuntu"
+        elif 'Red Hat' in (res[0] or res[1]):
+            self.distro = 'rhel'
+        elif 'SLES' in (res[0] or res[1]):
+            self.distro = 'sles'
+        else:
+            raise self.skipTest("Test currently supported only on ubuntu, sles and rhel")
+
+ 
+    def kexec_load(self):
+        """
+        This function will test the kexec load
+        """
+
+        try:
+            self.c.run_command("kexec -l /boot/{} --initrd /boot/{} --append=\"`cat /proc/cmdline`\"".format(self.kernel_image,self.initrd_image))
+        except CommandFailed:
+            kexec_loaded_status = self.c.run_command("cat /sys/kernel/kexec_loaded")
+            if kexec_loaded_status[0] == "0":
+                log.info("New kernel not loaded- Failed with kexec_loaded_status = \"%s\"", kexec_loaded_status[0])
+            raise OpTestError("kexec -l failed to load kernel")
+        
+    def kexec_unload(self):
+        """
+        This function will test the kexec unload
+        """
+
+        try:
+            self.c.run_command("kexec -u")
+        except CommandFailed:
+            kexec_loaded_status = self.c.run_command("cat /sys/kernel/kexec_loaded")
+            if kexec_loaded_status[0] == "1":
+                log.info("kernel unload Failed with kexec_loaded_status = \"%s\"", kexec_loaded_status[0])
+            raise OpTestError("Kexec -u  failed to unload")
+        
+    def kexec_load_reboot(self):
+        """
+        This function will test the kexec load and reboot
+        """
+
+        try:
+            self.c.run_command("kexec -l /boot/{} --initrd /boot/{} --append=\"`cat /proc/cmdline`\"".format(self.kernel_image,self.initrd_image))
+        except CommandFailed:
+            kexec_loaded_status = self.c.run_command("cat /sys/kernel/kexec_loaded")
+            if kexec_loaded_status[0] == "0":
+                log.info("New kernel not loaded- Failed with kexec_loaded_status = \"%s\"", kexec_loaded_status[0])
+            raise OpTestError("kexec -l failed to load kernel")
```

Comment:
Instead of replicating the code here can kexec_load() function be called first and them attempt to run kexec -e?

Reply:
done